confirm.c

```diff
@@ -763,15 +857,19 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_tcp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
+		uint16_t src_port, uint16_t dst_port, confirm_query_cb cb)
 {
-	struct confirm_query *query;
+	struct confirm_query *query = malloc(sizeof(struct confirm_query));
 
-	query = malloc(sizeof(struct confirm_query));
-	if(!query) logea(__FILE__, __LINE__, NULL);
+	if(!query){
```

Comment:
my argument is that error checking is boring and happens all the time; so I try to reduce the amount of space it takes int he code.